provider/service_data.go

```diff
@@ -132,22 +120,29 @@ func dataFireHydrantServices(ctx context.Context, d *schema.ResourceData, m inte
 
 	services := make([]interface{}, 0)
 
+	// Set the data source attributes to the values we got from the API
 	for _, svc := range r.Services {
 		values := map[string]interface{}{
 			"id":           svc.ID,
-			"name":         svc.Name,
+			"alert_on_add": svc.AlertOnAdd,
 			"description":  svc.Description,
+			"name":         svc.Name,
 			"service_tier": svc.ServiceTier,
-			"alert_on_add": svc.AlertOnAdd,
 		}
+
+		// Process any attributes that could be nil
+		if svc.Owner != nil {
+			values["owner_id"] = svc.Owner.ID
+		}
+
 		services = append(services, values)
 	}
-
 	if err := d.Set("services", services); err != nil {
 		return diag.FromErr(err)
 	}
 
 	var ds diag.Diagnostics
+
 	d.SetId("does-not-matter")
```

Comment:
Should we have an issue for doing something about this?

Reply:
💯 i haven't seen it cause problems yet in my testing but i'd still prefer it have a unique ID. 

i've got this and a few other refactoring tasks written down but not added to the issues yet. i'm hoping to get them in there at the end of this week.